client/src/lib/CKANApi/DataStore/Query.js

```diff
@@ -156,6 +156,43 @@ class Query {
    * @return {string}
    */
   parse(resource) {
+    const { distinct, fields, join, where, order, limit } = this._parseClauses(resource);
+
+    // Combine all the clauses after "FROM {table}" into one const
+    const endClause = where + order + limit;
+
+    // Combine it all together
+    return `SELECT ${distinct}${fields} FROM "${resource}"${join}${endClause}`;
+  }
+
+  /**
+   * Parse this object into an SQL statement that will count all records and can be sent through to
+   * a CKAN API endpoint. This includes distinct and filter options but ignores limit and offset.
+   *
+   * @param {string} resource
+   * @return {string}
+   */
+  parseCount(resource) {
+    const { join, where } = this._parseClauses(resource);
+
+    return `SELECT count(*) FROM "${resource}"${join}${where}`;
+  }
+
+  /**
+   * Parse various SQL clauses from the state of this object
+   *
+   * @param resource
+   * @return {{
+   *   limit: string,
+   *   distinct: string,
+   *   where: string,
+   *   join: string,
+   *   fields: string,
+   *   order: string
+   * }}
```

Comment:
Is this valid jsdoc syntax? If so, TIL

Reply:
Yeah to be honest I'm not sure. My IDE suggested it but I couldn't find any specification on the JS doc website. I did find some other comments about using it though. I'm happy to revert back to `{Object}`.